### DISCUSSION TITLE
Support 'Features' from the GeoJSON spec to store address data in the Map Field

### DISCUSSION THREAD

**Discussion by DannyCare** [directus/directus#22919] [Feature Requests]
### Summary

Currently, the Directus Map Field does not support storing address data using ‘Features’ from the [GeoJSON spec](https://geojson.readthedocs.io/en/latest/index.html?highlight=Features#feature). This feature request proposes adding support for ‘Features’ to enhance the flexibility and usability of the Map Field.

### Basic Example

```json
{
  "type": "Feature",
  "geometry": {
    "type": "Point",
    "coordinates": [125.6, 10.1]
  },
  "properties": {
    "street": "22 Rue du Grenier Saint-Lazare",
    "postalCode": "75003",
    "city": "Paris",
    "countryCode": "FRA",
    "country": "France",
    "name": "22 Rue du Grenier Saint-Lazare\n75003 Paris\nFrance"
  }
}
```

### Motivation

Adding support for ‘Features’ from the GeoJSON spec allows for richer data representation in the Map Field. This enhancement supports use cases where address data needs to be stored and displayed along with geographical coordinates. The expected outcome is a more versatile Map Field that can handle complex data structures, thereby improving data management and user experience.

This feature reduces the need for dedicated input fields for each location attribute and allows for multiple locations within a single field. All data about a location can be requested in one API call, instead of requiring separate calls (using Flows / Extensions, or from a Frontend application).

### Detailed Design

The design involves extending the current Map Field to parse and store GeoJSON ‘Features’. This requires:

1. Updating the schema to accept GeoJSON ‘Feature’ objects.
2. Modifying the input component to support ‘Feature’ creation and editing.
3. Ensuring the display component can render ‘Features’ correctly, including displaying address data.

### Must Have:

- [ ] Ability to store GeoJSON ‘Feature’ objects in the database.
- [ ] Display functionality for ‘Features’ including address data.

### Should Have:

- [ ] 	Validation to ensure correct GeoJSON format.
- [ ] 	Search and filter functionality based on address data.

### Could Have:
-

### Won't Have:
- [ ] 	Support for other GeoJSON types beyond ‘Point’ at this stage.

### Drawbacks

- Impact on existing features and planned integrations.
- Users migrating existing applications if changes are breaking.

### Alternatives

**Alternatives include:**

- Continuing with the current implementation and using custom fields for address data.
- Implementing a separate field type for address data.

**Impact of not doing this:**

- Continued limitation in the functionality of the Map Field.
- Increased complexity for users needing to store and display address data.

### Adoption Strategy

This can be introduced as a non-breaking change as it adds to the existing data structure without modifying any existing fields. Existing applications can continue using the current Map Field without modification. If users wanted to retroactively get this information they would have to update their item, or we can include an optional migration script which will make the calls to MapBox as part of the update.

### Unresolved Questions

- The performance impact of storing and rendering complex GeoJSON objects.
- User feedback on the usability of the new input methods for ‘Features’.

**Comment by sookoll** [comment]
I think You somewhat misunderstand GeoJSON. But I'm agree, that directus should take advantage of using GeoJSON as output/input format in general. Geometry is just a one field of others. All the map field thing in directus is just awful solution to everybody who has at least some experience with spatial data/development.

**Reply by rijkvanzanten** [reply]
> All the map field thing in directus is just awful solution to everybody who has at least some experience with spatial data/development.

Mind sharing thoughts on improvements instead of just "it's awful"? I am not one of the everybody that has spatial experience

**Reply by sookoll** [reply]
np. Lack of projection support. Lack of strategy how to handle large amount of data in map. Paging by count of rows is not a really thing in spatial world. As mentioned already, no standard spatial data format (GeoJSON, TopoJSON, GML etc.) output. True, spatial field contain GeoJSON geometry, one point for that. I'm not going into spatial data processing, as this would be next level. But some sane bbox filter should be implemented, if You want to handle spatial data on map. In map preset You can't set fields, which means You can't classify/style features on map based on some (set of) fields. Force users to use only Mapbox (paid service) for admin interface map is insanely stupid (instead OpenLayers or Leaflet map with OSM layer). Or am I wrong?

**Comment by jeremyzilar** [comment]
I am planning out how I'll store address information and geo location info and this would be very helpful!
